custom-metrics-stackdriver-adapter/pkg/adapter/translator/query_builder.go

```diff
@@ -37,17 +37,28 @@ var (
 	// allowedExternalMetricsLabelPrefixes and allowedExternalMetricsFullLabelNames specify all metric labels allowed for querying
 	// External Metrics API.
 	allowedExternalMetricsLabelPrefixes  = []string{"metric.labels", "resource.labels", "metadata.system_labels", "metadata.user_labels"}
-	allowedExternalMetricsFullLabelNames = []string{"resource.type"}
+	allowedExternalMetricsFullLabelNames = []string{"resource.type", "reducer"}
 	// allowedCustomMetricsLabelPrefixes and allowedCustomMetricsFullLabelNames specify all metric labels allowed for querying
 	allowedCustomMetricsLabelPrefixes  = []string{"metric.labels"}
-	allowedCustomMetricsFullLabelNames = []string{}
+	allowedCustomMetricsFullLabelNames = []string{"reducer"}
+
+	percentilesToReducers = map[string]string{
+		p99: "REDUCE_PERCENTILE_99",
```

Comment:
Are these 3 specifically defined at the back end or can you have p90 as well for example?

Reply:
Defined at the back end: https://cloud.google.com/monitoring/api/ref_v3/rest/v3/projects.alertPolicies#reducer.

I forgot about `REDUCE_PERCENTILE_05` however. Added now.